module/VuFind/config/module.config.php

```diff
@@ -378,7 +379,10 @@
             'VuFind\Crypt\HMAC' => 'VuFind\Crypt\HMACFactory',
             'VuFind\Date\Converter' => 'VuFind\Service\DateConverterFactory',
             'VuFind\Db\AdapterFactory' => 'VuFind\Service\ServiceWithConfigIniFactory',
+            'VuFind\Db\Connection' => 'VuFind\Db\ConnectionFactory',
```

Comment:
If the real class name is used, other connections will use the factory too. We could always create an empty `VuFind\Db\Connection` marker class inheriting `Doctrine\DBAL\Connection`?

Reply:
The general pattern is that the name of the service should match the resource created for the service. Can we use a base class or interface for Doctrine drivers, or are you saying that doing so would override core Doctrine configuration and cause problems?

Reply:
Yes, using `'Doctrine\DBAL\Connection' => 'VuFind\Db\ConnectionFactory'` causes the VuFind specific factory to be used for all Doctrine configuration based connections created by `AbstractDoctrineServiceFactory`. Other connections can be specifically configured to avoid this, but since VuFind has a connection created by a custom factory class maybe it should have a custom connection class as well.

Reply:
True, that does make sense.

Reply:
Will have to stop now for today but will push the new connection class tomorrow!